swarm/test/cli_test.go

```diff
@@ -0,0 +1,84 @@
+package swarmtest
+
+import (
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"regexp"
+	"strings"
+	"testing"
+)
+
+// TestCLISwarmUp tests that running 'swarm up' makes the resulting file
+// available from all nodes via the HTTP API
+func TestCLISwarmUp(t *testing.T) {
```

Comment:
These tests are basically just CLI tests yes, which I don't think will be covered by the scenario tests (they most likely won't be calling `swarm up`, `swarm manifest` etc.).

My vision is that we have three types of tests:

* Unit tests - testing the behaviour of individual functions / objects
* Integration tests - testing the behaviour from the end user's perspective using compiled binaries / HTTP / IPC requests
* Network scenario tests - testing how the nodes interact with each other at a large scale, running either in-memory / rlpx with unique ports / docker containers / kvm

WDYT?